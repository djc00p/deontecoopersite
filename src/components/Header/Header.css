.My-Site-header {
  margin: auto;
  display: flex;
  flex-direction: column;
  background-color: #901010;
  color: #ffffff;
  border-bottom: 0.25rem solid black;
}
/* Name and Occupation */
.yoursTruly {
  display: flex;
  flex-direction: column;
  margin: 1rem;
}
.name {
  font-size: 5rem;
  text-align: center;
}
.occupation {
  font-size: 2.5rem;
}
.center-piece {
  display: flex;
  justify-content: center;
}
/* NavBar */
nav {
  max-width: 40rem;
  padding: 0;
  height: 2rem;
  display: flex;
  position: relative;
}
.nav {
  min-width: 8.5rem;
  display: flex;
  justify-content: center;
  text-decoration: none;
  font-size: 1.5rem;
  color: inherit;
}
a:hover {
  color: black;
}
hr {
  height: 1rem;
  color: black;
  border: 1px solid black;
}
.navbar {
  width: auto;
  background-color: #901010;
}
/* Mobile NavBar */
.navbar-toggle {
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 100vw;
  font-size: 0.875rem;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji",
    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  text-decoration: none;
  background-color: inherit;
  border: 0;
  outline: 0;
  cursor: pointer;
}
.navbar-toggle:hover {
  background-color: #bb2020;
}
/* Dropdown */
.dropdown {
  position: relative;
  display: block;
}
.navbar-dropdown {
  display: none;
  position: relative;
  z-index: 1020;
  box-sizing: border-box;
  width: inherit;
  padding: 5px;
  background: #fff;
  color: black;
  box-shadow: 0 5px 12px rgba(0, 0, 0, 0.15);
}
.navbar-dropdown ul {
  padding: 0;
}
.navbar-dropdown a {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
}
.dropdown-divider {
  height: 0;
  width: 80%;
  border-top: 1px black solid;
}
.navbar-dropdown a:hover {
  background-color: #f1f1f1;
  width: 75%;
  color: #bb2020;
}
/* Show */
.show {
  display: block;
}
/* ========================================================================
   Component: Visibility
 ========================================================================== */
/*
 * Hidden
 * `hidden` attribute also set here to make it stronger
 */
[hidden],
.hidden {
  display: none !important;
}
/* Phone portrait and bigger */
@media (min-width: 640px) {
  .hidden\@m {
    display: none !important;
  }
}
/*
 * Visible
 */
/* Phone portrait and smaller */
@media (max-width: 639px) {
  .visible\@m {
    display: none !important;
  }
}
