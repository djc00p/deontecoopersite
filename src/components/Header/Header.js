import React from "react";
import { NavLink } from "react-router-dom";
import "./Header.css";

const Header = () => {
  const displayDropdownMenu = () => {
    document.getElementById("myDropdown").classList.toggle("show");
  };

  window.onclick = function(event) {
    if (
      !event.target.matches(".navbar-toggle") &&
      !event.target.matches(".menu-svg")
    ) {
      var dropdowns = document.getElementsByClassName("navbar-dropdown");
      var i;
      for (i = 0; i < dropdowns.length; i++) {
        var openDropdown = dropdowns[i];
        if (openDropdown.classList.contains("show")) {
          openDropdown.classList.remove("show");
        }
      }
    }
  };

  return (
    <header className="My-Site-header">
      <div className="yoursTruly">
        <span className="name center-piece" alt="Name">
          Deont√© J Cooper
        </span>
        <span className="occupation center-piece" alt="Occupation">
          Software Engineer
        </span>
      </div>
      <div className="navbar center-piece">
        <nav className="center-piece visible@m" alt="Navbar">
          <NavLink to="/bio" className="nav" alt="Bio Link">
            {" "}
            Bio{" "}
          </NavLink>
          <hr></hr>
          <NavLink to="/projects" className="nav" alt="Projects Link">
            {" "}
            Projects{" "}
          </NavLink>
          <hr></hr>
          <NavLink to="/soundcloud" className="nav" alt="SoundCloud Link">
            {" "}
            SoundCloud{" "}
          </NavLink>
          <hr></hr>
          <NavLink to="/blog" className="nav" alt="Blog Link">
            {" "}
            Blog{" "}
          </NavLink>
          <hr></hr>
          <NavLink to="/contact" className="nav" alt="Contact Link">
            {" "}
            Contact{" "}
          </NavLink>
        </nav>
        <nav className="center-piece hidden@m dropdown" alt="Navbar">
          <button
            onClick={displayDropdownMenu}
            className="navbar-toggle"
            alt="Navigation Dropdown Menu Button"
          >
            <svg
              width="60"
              height="30"
              viewBox="0 0 60 40"
              xmlns="http://www.w3.org/2000/svg"
              className="menu-svg"
            >
              <rect
                y="9"
                width="60"
                height="6"
                style={{ fill: "rgb(255,255,255)" }}
                className="menu-svg"
              ></rect>
              <rect
                y="18"
                width="60"
                height="6"
                style={{ fill: "rgb(255,255,255)" }}
                className="menu-svg"
              ></rect>
              <rect
                y="27"
                width="60"
                height="6"
                style={{ fill: "rgb(255,255,255)" }}
                className="menu-svg"
              ></rect>
            </svg>
          </button>
          <div
            id="myDropdown"
            className="navbar-dropdown"
            alt="Navigation Dropdown Menu"
          >
            <ul>
              <li className="nav">
                <NavLink to="/bio" className="nav" alt="Bio Link">
                  {" "}
                  Bio{" "}
                </NavLink>
              </li>
              <hr className="dropdown-divider"></hr>
              <li className="nav">
                <NavLink to="/projects" className="nav" alt="Projects Link">
                  {" "}
                  Projects{" "}
                </NavLink>
              </li>
              <hr className="dropdown-divider"></hr>
              <li className="nav">
                <NavLink to="/soundcloud" className="nav" alt="SoundCloud Link">
                  {" "}
                  SoundCloud{" "}
                </NavLink>
              </li>
              <hr className="dropdown-divider"></hr>
              <li className="nav">
                <NavLink to="/blog" className="nav" alt="Blog Link">
                  {" "}
                  Blog{" "}
                </NavLink>
              </li>
              <hr className="dropdown-divider"></hr>
              <li className="nav">
                <NavLink to="/contact" className="nav" alt="Contact Link">
                  {" "}
                  Contact{" "}
                </NavLink>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </header>
  );
};

export default Header;
