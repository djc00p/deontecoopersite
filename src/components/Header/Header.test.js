import React from "react";
import { shallow, mount, render } from "enzyme";
import Header from "./Header";

describe("Header", () => {
  it("should create snapshot of header", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
