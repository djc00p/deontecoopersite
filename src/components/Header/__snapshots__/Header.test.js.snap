// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should create snapshot of header 1`] = `
<header
  className="My-Site-header"
>
  <div
    className="yoursTruly"
  >
    <span
      alt="Name"
      className="name center-piece"
    >
      Deont√© J Cooper
    </span>
    <span
      alt="Occupation"
      className="occupation center-piece"
    >
      Software Engineer
    </span>
  </div>
  <div
    className="navbar center-piece"
  >
    <nav
      alt="Navbar"
      className="center-piece visible@m"
    >
      <NavLink
        alt="Bio Link"
        className="nav"
        to="/bio"
      >
         
        Bio
         
      </NavLink>
      <hr />
      <NavLink
        alt="Projects Link"
        className="nav"
        to="/projects"
      >
         
        Projects
         
      </NavLink>
      <hr />
      <NavLink
        alt="Blog Link"
        className="nav"
        to="/blog"
      >
         
        Blog
         
      </NavLink>
      <hr />
      <NavLink
        alt="Contact Link"
        className="nav"
        to="/contact"
      >
         
        Contact
         
      </NavLink>
    </nav>
    <nav
      alt="Navbar"
      className="center-piece hidden@m dropdown"
    >
      <button
        alt="Navigation Dropdown Menu Button"
        className="navbar-toggle"
        onClick={[Function]}
      >
        <svg
          className="menu-svg"
          height="30"
          viewBox="0 0 60 40"
          width="60"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            className="menu-svg"
            height="6"
            style={
              Object {
                "fill": "rgb(255,255,255)",
              }
            }
            width="60"
            y="9"
          />
          <rect
            className="menu-svg"
            height="6"
            style={
              Object {
                "fill": "rgb(255,255,255)",
              }
            }
            width="60"
            y="18"
          />
          <rect
            className="menu-svg"
            height="6"
            style={
              Object {
                "fill": "rgb(255,255,255)",
              }
            }
            width="60"
            y="27"
          />
        </svg>
      </button>
      <div
        alt="Navigation Dropdown Menu"
        className="navbar-dropdown"
        id="myDropdown"
      >
        <ul>
          <li
            className="nav"
          >
            <NavLink
              alt="Bio Link"
              className="nav"
              to="/bio"
            >
               
              Bio
               
            </NavLink>
          </li>
          <hr
            className="dropdown-divider"
          />
          <li
            className="nav"
          >
            <NavLink
              alt="Projects Link"
              className="nav"
              to="/projects"
            >
               
              Projects
               
            </NavLink>
          </li>
          <hr
            className="dropdown-divider"
          />
          <li
            className="nav"
          >
            <NavLink
              alt="Blog Link"
              className="nav"
              to="/blog"
            >
               
              Blog
               
            </NavLink>
          </li>
          <hr
            className="dropdown-divider"
          />
          <li
            className="nav"
          >
            <NavLink
              alt="Contact Link"
              className="nav"
              to="/contact"
            >
               
              Contact
               
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</header>
`;
